src/visuals/settings_menu.rs

```diff
@@ -0,0 +1,693 @@
+                            spinner_buttons_container.spawn(volume_slider.spinner.increment);
+                            spinner_buttons_container.spawn(volume_slider.spinner.decrement);
+                        });
+                });
+        });
+}
+
+/// Creates a Counting Slider with a Label.
+fn create_counting_slider(label: String) -> CountingSlider {
+    let slider_widget_keys = SliderKeyComponents { array: [None; 6] };
+
+    let slider_widget_text = create_widget_label(label);
+    let slider = create_widget_slider();
+    let spinner = create_widget_spinner();
+
+    CountingSlider {
+        barrier_keys: slider_widget_keys,
+
+        label: slider_widget_text,
+        slider,
+        spinner,
+    }
+}
+
+fn create_button(b_type: ButtonTypes) -> (ButtonBundle, ButtonTypes, SettingsMenuElements) {
+    (
+        ButtonBundle {
+            style: Style {
+                width: Val::Percent(11.33),
+                height: Val::Percent(40.0),
+                justify_content: JustifyContent::Center,
+                align_items: AlignItems::Center,
+                ..default()
+            },
+            ..default()
+        },
+        b_type,
+        SettingsMenuElements::Button,
+    )
+}
+
+fn create_button_text(text: String) -> (TextBundle, SettingsMenuElements) {
+    (
+        TextBundle::from_section(
+            text,
+            TextStyle {
+                font_size: 20.0,
+                color: WHITE,
+                ..default()
+            },
+        ),
+        SettingsMenuElements::Text,
+    )
+}
+
+fn create_widget_container(keys: SliderKeyComponents) -> (NodeBundle, SliderKeyComponents) {
+    (
+        NodeBundle {
+            style: Style {
+                width: Val::Percent(96.0),
+                height: Val::Percent(12.0),
+                justify_content: JustifyContent::SpaceBetween,
+                ..default()
+            },
+            ..default()
+        },
+        keys,
+    )
+}
+
+pub fn change_music_volume(
+    spinner_query: Query<(&Text, &AudioType), Changed<Text>>,
+    background_music: Res<AudioChannel<MusicChannel>>,
+) {
+    for (text, audio_type) in &spinner_query {
+        if *audio_type != AudioType::Music {
+            continue;
+        }
+
+        let slider_percentage = get_percentage_from(text.clone());
+
+        background_music.set_volume(slider_percentage);
+    }
+}
+
+pub fn change_sfx_volumes(
+    spinner_query: Query<(&Text, &AudioType), Changed<Text>>,
+    player_movement_sound: Res<AudioChannel<PlayerWalkChannel>>,
+    player_bump_sound: Res<AudioChannel<PlayerBumpChannel>>,
+) {
+    for (text, slider_type) in &spinner_query {
+        if *slider_type != AudioType::SFX {
+            continue;
+        }
+
+        let slider_percentage = get_percentage_from(text.clone());
+
+        player_movement_sound.set_volume(slider_percentage);
+        player_bump_sound.set_volume(slider_percentage);
+    }
+}
```

Comment:
If we want to be consistent between setting up the UI (the visuals module, which we're in right now) versus reacting to the UI (the mechanics module), what do you think about moving these two systems into the mechanics module relating to the settings menu?

Reply:
We'll address this in a new issue, this is one of many functions that needs to be relocated. The project needs general reorganizing due to how large this feature is.